### ISSUE TITLE
pl-template examples are inaccessible

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#11444]
Lifting this out of https://github.com/PrairieLearn/PrairieLearn/pull/11429#discussion_r1968162056 and https://github.com/PrairieLearn/PrairieLearn/pull/11429#discussion_r1968171269.

The following examples provided for `<pl-template>` aren't accessible. They rely on clicks on non-focusable elements and lack proper `aria-*` attributes.

https://github.com/PrairieLearn/PrairieLearn/blob/450366e9689d20f8478b52e99a1e77ef392b2ad7/exampleCourse/serverFilesCourse/templates/collapsible_section.mustache#L3-L20

https://github.com/PrairieLearn/PrairieLearn/blob/450366e9689d20f8478b52e99a1e77ef392b2ad7/docs/elements.md?plain=1#L1781-L1800

Per discussion with @eliotwrobson, we should replace these examples with something simpler, probably just a basic reimplementation of `<pl-card>`.
